### DISCUSSION TITLE
where is the main file?

### DISCUSSION THREAD

**Discussion by exosane** [Cog-Creators/Red-DiscordBot#4796] [Q&A] [answered]
pls help

**Comment by Drapersniper** [comment]
First could you explain what are you trying to achive by locating the "main" file?

**Comment by Apoo711** [comment, marked as answer]
"Red does not have a main python file in the way that you are asking. If your host is asking for your bot's main file, it is unsupported by Red and will likely not work with our data structure or program, due to the fact that these hosts aren't proper VPSs. Take a look at our host list https://docs.discord.red/en/stable/host-list.html to see a list of the recommended providers that offer actual VPSes"
